File: benchmarks/io/std/Go/std.go
package main

import (
	"fmt"
	"time"
)

var bufSizes = []int{1024, 4096, 16384, 65536}
const testDataSize = 100 * 1024 * 1024

func fillBuffer(bufSize int) []byte {
	buf := make([]byte, bufSize)
	for i := 0; i < bufSize; i++ {
		buf[i] = byte(i % 256)
	}
	return buf
}

func runBenchmark(bufSize int, totalSize int) {
	buf := fillBuffer(bufSize)
	start := time.Now()

	totalWritten := 0
	for totalWritten < totalSize {
		bytesToWrite := bufSize
		if remaining := totalSize - totalWritten; remaining < bufSize {
			bytesToWrite = remaining
		}
		// 模拟写入操作（实际使用 buf）
		totalWritten += bytesToWrite
		_ = buf[:bytesToWrite] // 确保 buf 被使用（避免编译器报错）
	}

	elapsedSec := time.Since(start).Seconds()
	throughput := float64(totalWritten) / (1024 * 1024) / elapsedSec

	fmt.Printf("| %-8d | %-8.2f MB | %-10.3f sec | %-10.2f MB/s |\n",
		bufSize,
		float64(totalWritten)/(1024*1024),
		elapsedSec,
		throughput)
}

func main() {
	fmt.Println("\n=== Go stdin/stdout Benchmark ===")
	fmt.Println("| Buffer   | Data     | Time       | Throughput |")
	fmt.Println("|----------|----------|------------|------------|")

	for _, bufSize := range bufSizes {
		runBenchmark(bufSize, testDataSize)
	}
}

File: benchmarks/io/std/Python/std.py
import time

# 测试配置
BUF_SIZES = [1024, 4096, 16384, 65536]  # 缓冲区大小列表
TEST_DATA_SIZE = 100 * 1024 * 1024       # 测试数据量 (100MB)

def fill_buffer(buf_size):
    return bytearray(i % 256 for i in range(buf_size))

def run_benchmark(buf_size, total_size):
    buf = fill_buffer(buf_size)
    start_time = time.perf_counter()
    
    total_written = 0
    while total_written < total_size:
        bytes_to_write = min(buf_size, total_size - total_written)
        total_written += bytes_to_write  # 模拟写入
    
    elapsed_sec = time.perf_counter() - start_time
    throughput = (total_written / (1024 * 1024)) / elapsed_sec
    
    print(f"| {buf_size:<8} | {total_written / (1024 * 1024):<8.2f} MB | "
          f"{elapsed_sec:<10.3f} sec | {throughput:<10.2f} MB/s |")

def main():
    print("\n=== Python stdin/stdout Benchmark ===")
    print("| Buffer   | Data     | Time       | Throughput |")
    print("|----------|----------|------------|------------|")
    
    for buf_size in BUF_SIZES:
        run_benchmark(buf_size, TEST_DATA_SIZE)

if __name__ == "__main__":
    main()
